core/src/main/java/org/testcontainers/utility/RegistryAuthLocator.java

```diff
@@ -162,15 +163,43 @@ private Optional<AuthConfig> lookupUncachedAuthConfig(String registryName, Docke
             }
         } catch (Exception e) {
             log.info(
-                "Failure when attempting to lookup auth config. Please ignore if you don't have images in an authenticated registry. Details: (dockerImageName: {}, configFile: {}. Falling back to docker-java default behaviour. Exception message: {}",
+                "Failure when attempting to lookup auth config. Please ignore if you don't have images in an authenticated registry. Details: (dockerImageName: {}, configFile: {}, configEnv: {}). Falling back to docker-java default behaviour. Exception message: {}",
                 dockerImageName,
                 configFile,
+                DOCKER_AUTH_ENV_VAR,
                 e.getMessage()
             );
         }
         return Optional.empty();
     }
 
+    private JsonNode getDockerAuthConfig() throws Exception {
+        log.debug(
+            "RegistryAuthLocator has configFile: {} ({}) configEnv: {} ({}) and commandPathPrefix: {}",
+            configFile,
+            configFile.exists() ? "exists" : "does not exist",
+            DOCKER_AUTH_ENV_VAR,
+            configEnv != null ? "exists" : "does not exist",
+            commandPathPrefix
+        );
+
+        if (configFile.exists()) {
+            log.debug("RegistryAuthLocator reading from configFile: {}", configFile);
+            return OBJECT_MAPPER.readTree(configFile);
+        } else if (configEnv != null) {
```

Comment:
Shouldn't the env var be evaluated first? Usually, you can use env vars to override config files but this would not be the case here (if you have both the conf file and the env var). Maybe I'm missing something here?

Reply:
good catch! yes, you're right.

Reply:
@roseo1 can you take care of this, please? If not, LMK so I can do it before merge it.